### PR TITLE
[main][fix] strategy partial close liquidate

### PR DESCRIPTION
**Pull request by jakeal-papaca**
## Description
 - add event to PancakeswapV2RestrictedStrategyPartialCloseLiquidate and WaultSwapRestrictedStrategyPartialCloseLiquidate contract

## Changelogs
 - PancakeswapV2RestrictedStrategyPartialCloseLiquidate, WaultSwapRestrictedStrategyPartialCloseLiquidate contract
 - unit test

### CODE REVIEW

File: contracts/6/protocol/strategies/waultswap-restricted/WaultSwapRestrictedStrategyPartialCloseLiquidate.sol [hunk 1]
```diff
@@ -60,20 +67,24 @@ contract WaultSwapRestrictedStrategyPartialCloseLiquidate is OwnableUpgradeSafe,
     bytes calldata data
   ) external override onlyWhitelistedWorkers nonReentrant {
     // 1. Find out what farming token we are dealing with.
-    (uint256 returnLpToken, uint256 minBaseToken) = abi.decode(data, (uint256, uint256));
+    (uint256 lpTokenToLiquidate, uint256 toRepaidBaseTokenDebt, uint256 minBaseToken) =
```
Comment (spicysquid168) [inline contracts/6/protocol/strategies/waultswap-restricted/WaultSwapRestrictedStrategyPartialCloseLiquidate.sol:70]: What is the use of `toRepaidBaseTokenDebt` except emit event?

File: test/PancakeswapV2_Restricted_StrategyPartialCloseLiquidate.test.ts [hunk 2]
```diff
@@ -256,16 +256,16 @@ describe('PancakeswapV2RestrictedStrategyPartialCloseLiquidate', () => {
...
               ['address', 'bytes'],
               [strat.address, ethers.utils.defaultAbiCoder.encode(
-              ['uint256', 'uint256'],
-              [returnLp, ethers.utils.parseEther('0.5')]
+              ['uint256', 'uint256', 'uint256'],
+              [returnLp, ethers.utils.parseEther('0.1'), ethers.utils.parseEther('0.5')]
             )],
           )
-      )
+      )).to.emit(strat, 'PancakeswapV2RestrictedStrategyPartialCloseLiquidateEvent')
```
Comment (spicysquid168) [inline test/PancakeswapV2_Restricted_StrategyPartialCloseLiquidate.test.ts:268]: expect args too

File: test/WaultSwap_Restricted_StrategyPartialCloseLiquidate.test.ts [hunk 3]
```diff
@@ -255,16 +255,16 @@ describe('WaultSwapRestrictedStrategyPartialCloseLiquidate', () => {
...
               ['address', 'bytes'],
               [strat.address, ethers.utils.defaultAbiCoder.encode(
-              ['uint256', 'uint256'],
-              [returnLp, ethers.utils.parseEther('0.5')]
+              ['uint256', 'uint256', 'uint256'],
+              [returnLp, ethers.utils.parseEther('0.1'), ethers.utils.parseEther('0.5')]
             )],
           )
-      )
+      )).to.emit(strat, 'WaultSwapRestrictedStrategyPartialCloseLiquidateEvent')
```
Comment (spicysquid168) [inline test/WaultSwap_Restricted_StrategyPartialCloseLiquidate.test.ts:267]: expect args from event too

**Review by spicysquid168 — CHANGES_REQUESTED** [review]

File: contracts/6/protocol/strategies/waultswap-restricted/WaultSwapRestrictedStrategyPartialCloseLiquidate.sol (hunk 1 above)
Comment (jakeal-papaca) [reply contracts/6/protocol/strategies/waultswap-restricted/WaultSwapRestrictedStrategyPartialCloseLiquidate.sol:70]: only emit event

File: contracts/6/protocol/strategies/pancakeswapV2-restricted/PancakeswapV2RestrictedStrategyPartialCloseLiquidate.sol [hunk 4]
```diff
@@ -50,40 +64,53 @@ contract PancakeswapV2RestrictedStrategyPartialCloseLiquidate is OwnableUpgradeS
...
+    uint256, /* debt */
+    bytes calldata data
+  ) external override onlyWhitelistedWorkers nonReentrant {
     // 1. Find out what farming token we are dealing with.
-    (
-      uint256 returnLpToken,
-      uint256 minBaseToken
-    ) = abi.decode(data, (uint256, uint256));
+    (uint256 lpTokenToLiquidate, uint256 toRepaidBaseTokenDebt, uint256 minBaseToken) =
+      abi.decode(data, (uint256, uint256, uint256));
```
Comment (spicysquid168) [inline contracts/6/protocol/strategies/pancakeswapV2-restricted/PancakeswapV2RestrictedStrategyPartialCloseLiquidate.sol:74]: add require(toRepaidBaseTokenDebt <= debt)

**Review by spicysquid168 — CHANGES_REQUESTED** [review]

**Review by spicysquid168 — DISMISSED** [review]

**Review by spicysquid168 — DISMISSED** [review]

**Review by sundayalpaca — DISMISSED** [review]

**Review by spicysquid168 — APPROVED** [review]

**Review by sundayalpaca — APPROVED** [review]

**Review by almost-llamas — APPROVED** [review]

**Pull request merged**